lib/Frontend/CompilerInvocation.cpp

```diff
@@ -47,7 +47,25 @@ static void updateRuntimeLibraryPath(SearchPathOptions &SearchPathOpts,
   llvm::sys::path::append(LibPath, getPlatformNameForTriple(Triple));
   SearchPathOpts.RuntimeLibraryPath = LibPath.str();
 
-  llvm::sys::path::append(LibPath, Triple.getArchName());
+  // The linux provided triple for ARM contains a trailing 'l'
+  // denoting little-endian.  This is not used in the path for
+  // libraries.  LLVM matches these SubArchTypes to the generic
+  // ARMSubArch_v7 (for example) type.  If that is the case,
+  // use the base of the architecture type in the library path.
+  if (Triple.isOSLinux()) {
```

Comment:
@jrose-apple Your comment got lost again in the I've pasted it here from my email:

> I'm a little concerned about this function mutating Triple. It should probably be passed const.

>LLVM does have logic to canonicalize a triple. If you do that early on in the Driver, does everything work?

And I think that's a good point.  I refactored it a bit, and I think that this solution really gets at the essence of what I'm trying to achieve here.  The goal is to get the library path using the architecture type.  In many other places in the project, "armv7" and "armv6" are used for this path according to the architecture type (not the least of which are the build and install scripts).  It probably makes sense to decouple the assumption that llvm uses the same string to achieve this, or the unnecessary setting of llvm::Triple's string to match.  I think it makes more sense to ask llvm for the architecture we're using, then use that to select the explicit string that we're really looking for, and is hard-coded elsewhere.

With regard to the normalization logic, I did some experiments with that.  It preserves the 'l' provided by linux, so it won't do what we want here.